servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,184 @@
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.DELETE_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntries(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.READ_ENTRIES);
+  }
+
+  @Override
+  public void canListCommitLog(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.LIST_COMMIT_LOG);
+  }
+
+  @Override
+  public void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.COMMIT_CHANGE_AGAINST_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.READ_ENTITY_VALUE);
+  }
+
+  @Override
+  public void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.UPDATE_ENTITY);
+  }
+
+  @Override
+  public void canDeleteEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.DELETE_ENTITY);
+  }
+
+  private void canPerformOpOnReference(
+      AccessContext context, NamedRef ref, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        compiledRules.getRules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return entry
+                        .getValue()
+                        .execute(
+                            Boolean.class,
+                            ImmutableMap.of(
+                                "ref",
+                                ref.getName(),
+                                "role",
+                                getRoleName(context),
+                                "op",
+                                type.name()));
+                  } catch (ScriptException e) {
+                    throw new RuntimeException(
+                        String.format(
+                            "Failed to execute query expression with id '%s' and expression '%s' due to: %s",
+                            entry.getKey(), entry.getValue(), e.getMessage()));
+                  }
+                });
+    if (!allowed) {
+      throw new AccessControlException(
+          String.format(
+              "'%s' is not allowed for Role '%s' on Reference '%s'",
+              type, getRoleName(context), ref.getName()));
+    }
+  }
+
+  private String getRoleName(AccessContext context) {
+    return null != context.user() ? context.user().getName() : "";
+  }
+
+  private void canPerformOpOnPath(
```

Comment:
Think, you can unify this method with `canPerformOpOnReference` by preparding the map passed to `execute` in advance, which would also be cheaper for the runtime, because all script-args are the same.

Reply:
done